
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>4. Aging Clocks &#8212; Computational aging book</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="5. Complex systems approach" href="../dyn/complex_systems.html" />
    <link rel="prev" title="3. Hazard models" href="../stat/hazard_models.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/symbol.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Computational aging book</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Welcome to Computational Biology of Aging book
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Biology of Aging
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../bio/intro_aging_biology.html">
   1. Introduction to Aging Biology
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Statistical view
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../stat/differential_analysis_practice.html">
   2. Differential expression analysis
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../stat/hazard_models.html">
   3. Hazard models
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Machine learning view
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   4. Aging Clocks
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Dynamics system view
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../dyn/complex_systems.html">
   5. Complex systems approach
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../dyn/system_resilience.html">
   6. System resilience
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/ComputationalAgingLab/computational_aging_course"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/ComputationalAgingLab/computational_aging_course/issues/new?title=Issue%20on%20page%20%2Fml/aging_clocks.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../_sources/ml/aging_clocks.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.md</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#what-is-it">
   4.1. What is it
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#types-of-biological-clocks">
   4.2. Types of Biological Clocks
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#epigenetics-clocks">
     4.2.1. Epigenetics Clocks
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#epigenetic-predictor-of-age">
       4.2.1.1. 2011 Epigenetic predictor of age
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#genome-wide-methylation-profiles-reveal-quantitative-views-of-human-aging-rates">
       4.2.1.2. 2013 Genome-wide methylation profiles reveal quantitative views of human aging rates
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#dna-methylation-age-of-human-tissues-and-cell-types">
       4.2.1.3. 2013 DNA methylation age of human tissues and cell types
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#an-epigenetic-biomarker-of-aging-for-lifespan-and-healthspan-phenoage">
       4.2.1.4. 2018 An epigenetic biomarker of aging for lifespan and healthspan (PhenoAge)
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#dna-methylation-grimage-strongly-predicts-lifespan-and-healthspan">
       4.2.1.5. 2019 DNA methylation GrimAge strongly predicts lifespan and healthspan
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#other-clocks">
     4.2.2. Other Clocks
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#transcriptomic-clocks">
       4.2.2.1. Transcriptomic Clocks
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#blood-test-clocks">
       4.2.2.2. Blood Test Clocks
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#application-of-biological-clocks">
   4.3. Application of Biological Clocks
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#open-questions">
     4.3.1. Open Questions
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#linear-regression">
   4.4. Linear Regression
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#overfitting-and-regularization">
     4.4.1. Overfitting and Regularization
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#l1-and-l2-regularization">
     4.4.2. L1 and L2 regularization
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#elasticnet">
     4.4.3. Elasticnet
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#conclusion">
   4.5. Conclusion
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#credits">
   4.6. Credits
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Aging Clocks</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#what-is-it">
   4.1. What is it
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#types-of-biological-clocks">
   4.2. Types of Biological Clocks
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#epigenetics-clocks">
     4.2.1. Epigenetics Clocks
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#epigenetic-predictor-of-age">
       4.2.1.1. 2011 Epigenetic predictor of age
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#genome-wide-methylation-profiles-reveal-quantitative-views-of-human-aging-rates">
       4.2.1.2. 2013 Genome-wide methylation profiles reveal quantitative views of human aging rates
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#dna-methylation-age-of-human-tissues-and-cell-types">
       4.2.1.3. 2013 DNA methylation age of human tissues and cell types
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#an-epigenetic-biomarker-of-aging-for-lifespan-and-healthspan-phenoage">
       4.2.1.4. 2018 An epigenetic biomarker of aging for lifespan and healthspan (PhenoAge)
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#dna-methylation-grimage-strongly-predicts-lifespan-and-healthspan">
       4.2.1.5. 2019 DNA methylation GrimAge strongly predicts lifespan and healthspan
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#other-clocks">
     4.2.2. Other Clocks
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#transcriptomic-clocks">
       4.2.2.1. Transcriptomic Clocks
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#blood-test-clocks">
       4.2.2.2. Blood Test Clocks
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#application-of-biological-clocks">
   4.3. Application of Biological Clocks
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#open-questions">
     4.3.1. Open Questions
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#linear-regression">
   4.4. Linear Regression
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#overfitting-and-regularization">
     4.4.1. Overfitting and Regularization
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#l1-and-l2-regularization">
     4.4.2. L1 and L2 regularization
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#elasticnet">
     4.4.3. Elasticnet
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#conclusion">
   4.5. Conclusion
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#credits">
   4.6. Credits
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="aging-clocks">
<h1><a class="toc-backref" href="#id112"><span class="section-number">4. </span>Aging Clocks</a><a class="headerlink" href="#aging-clocks" title="Permalink to this headline">#</a></h1>
<p>In this chapter, you will learn:</p>
<ul class="simple">
<li><p>How to conduct clinical trials for anti-aging drugs ten times cheaper and faster.</p></li>
<li><p>The results of the last ten years of research on aging clocks.</p></li>
<li><p>How to use aging clocks to find new in silico geroprotectors.</p></li>
<li><p>Which aging clocks are the most accurate. Which are the cheapest.</p></li>
<li><p>Why not everything is straightforward and what else needs to be done in this area.</p></li>
</ul>
<p>During the practice, you will create your own aging clocks using machine learning.</p>
<div class="contents topic" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#aging-clocks" id="id112">Aging Clocks</a></p>
<ul>
<li><p><a class="reference internal" href="#what-is-it" id="id113">What is it</a></p></li>
<li><p><a class="reference internal" href="#types-of-biological-clocks" id="id114">Types of Biological Clocks</a></p>
<ul>
<li><p><a class="reference internal" href="#epigenetics-clocks" id="id115">Epigenetics Clocks</a></p>
<ul>
<li><p><a class="reference internal" href="#epigenetic-predictor-of-age" id="id116">2011 Epigenetic predictor of age</a></p></li>
<li><p><a class="reference internal" href="#genome-wide-methylation-profiles-reveal-quantitative-views-of-human-aging-rates" id="id117">2013 Genome-wide methylation profiles reveal quantitative views of human aging rates</a></p></li>
<li><p><a class="reference internal" href="#dna-methylation-age-of-human-tissues-and-cell-types" id="id118">2013 DNA methylation age of human tissues and cell types</a></p></li>
<li><p><a class="reference internal" href="#an-epigenetic-biomarker-of-aging-for-lifespan-and-healthspan-phenoage" id="id119">2018 An epigenetic biomarker of aging for lifespan and healthspan (PhenoAge)</a></p></li>
<li><p><a class="reference internal" href="#dna-methylation-grimage-strongly-predicts-lifespan-and-healthspan" id="id120">2019 DNA methylation GrimAge strongly predicts lifespan and healthspan</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#other-clocks" id="id121">Other Clocks</a></p>
<ul>
<li><p><a class="reference internal" href="#transcriptomic-clocks" id="id122">Transcriptomic Clocks</a></p></li>
<li><p><a class="reference internal" href="#blood-test-clocks" id="id123">Blood Test Clocks</a></p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference internal" href="#application-of-biological-clocks" id="id124">Application of Biological Clocks</a></p>
<ul>
<li><p><a class="reference internal" href="#open-questions" id="id125">Open Questions</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#linear-regression" id="id126">Linear Regression</a></p>
<ul>
<li><p><a class="reference internal" href="#overfitting-and-regularization" id="id127">Overfitting and Regularization</a></p></li>
<li><p><a class="reference internal" href="#l1-and-l2-regularization" id="id128">L1 and L2 regularization</a></p></li>
<li><p><a class="reference internal" href="#elasticnet" id="id129">Elasticnet</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#conclusion" id="id130">Conclusion</a></p></li>
<li><p><a class="reference internal" href="#credits" id="id131">Credits</a></p></li>
</ul>
</li>
</ul>
</div>
<section id="what-is-it">
<h2><a class="toc-backref" href="#id113"><span class="section-number">4.1. </span>What is it</a><a class="headerlink" href="#what-is-it" title="Permalink to this headline">#</a></h2>
<p>Imagine that a genie gives you a magic device that shows people’s health as a number from zero to one hundred. For healthy young people, the device shows 100. Over time, the number decreases and when it reaches zero, the person dies. How could such a device be used?</p>
<p>First, it could speed up and cheapen clinical trials. If you are looking for a geroprotector, the primary endpoint of the trial is lifespan. People live on average 72 years, so it takes a lot of money and time to do the research, so you have all the chances to die yourself before you get the results. For example, now funds are being raised for the TAME <span id="id1">[<a class="reference internal" href="#id40" title="Nir Barzilai, Jill P Crandall, Stephen B Kritchevsky, and Mark A Espeland. Metformin as a tool to target aging. Cell metabolism, 23(6):1060–1065, 2016. URL: https://doi.org/10.1016/j.cmet.2016.05.011.">Barzilai <em>et al.</em>, 2016</a>]</span> clinical trial of metformin for aging. In the last seven years, TAME – Targeting Aging with Metformin – has raised only 11 million dollars out of the necessary 40, and patient observation will last 6 years. If we had such a magic device, we wouldn’t necessarily have to wait so long. Instead of the primary endpoint (prolonging lifespan), we could use a surrogate endpoint. If the device shows an improvement in health, this would be a good sign that the geroprotector is working and we could stop the trial earlier. This would save money and time.</p>
<p>Secondly, the magic device would help us search for new interventions against aging. For example, we could collect a large dataset of device readings for different people and try to find associations between lifestyle and better health. Or if our magic device is a computer program that predicts a person’s health based on their indicators, we could run an optimization process: we would select such indicators to maximize health, and then find a way to similarly change human indicators. We will see how this idea helped find new geroprotectors that extended the life of worms <span id="id2">[<a class="reference internal" href="#id41" title="Georges E Janssens, Xin-Xuan Lin, Lluís Millan-Ariño, Alan Kavšek, Ilke Sen, Renée I Seinstra, Nicholas Stroustrup, Ellen AA Nollen, and Christian G Riedel. Transcriptomics-based screening identifies pharmacological inhibition of hsp90 as a means to defer aging. Cell reports, 27(2):467–480, 2019. URL: https://doi.org/10.1016/j.celrep.2019.03.044.">Janssens <em>et al.</em>, 2019</a>]</span>.</p>
<p>This magical device, referred to as a <strong>biological clock</strong> or <strong>aging clock</strong>, has been the primary focus of research in the field of the biology of aging in the past decade. It is usually a statistical model trained to predict a person’s chronological age based on their parameters, such as blood tests. The model is not perfect, so the predicted number differs from the actual chronological age. This prediction is called <strong>biological age</strong> and it is assumed that the error of the model is explained by the person’s health. A sick smoking person has a higher biological age than chronological age, while a healthy athlete has a lower biological age. The difference between biological and chronological age is called <strong>biological age acceleration</strong>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p><strong>Chronological age</strong> is the number of years a person has been alive since their birth.</p></li>
<li><p><strong>Biological age</strong> is a number that signifies a person’s health. In practice, this is a prediction of a model trained to predict chronological age.</p></li>
<li><p><strong>Biological age acceleration</strong> is the difference between biological and chronological age.</p></li>
</ul>
</div>
</section>
<section id="types-of-biological-clocks">
<span id="bioclocks-type"></span><h2><a class="toc-backref" href="#id114"><span class="section-number">4.2. </span>Types of Biological Clocks</a><a class="headerlink" href="#types-of-biological-clocks" title="Permalink to this headline">#</a></h2>
<p>The <a class="reference external" href="https://ourworldindata.org/grapher/annual-number-of-deaths-by-cause">most common causes of death</a> in developed countries are cancer, cardiovascular disease, type 2 diabetes, Alzheimer’s disease, and Parkinson’s disease. These diseases have different mechanisms, but the strongest risk factor for all of them is age, so researchers of aging believe that aging is not just an association but a cause of many diseases, so slowing aging would reduce the likelihood of illness. This is called the <strong>geroscience hypothesis</strong>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>The <strong>geroscience hypothesis</strong> proposes that aging is the primary cause of many chronic diseases, and that these diseases can be delayed by slowing down aging.</p></li>
</ul>
</div>
<p>Aging biomarkers are an important focus of research in the field of aging because what cannot be measured cannot be improved. There are several key characteristics that are desirable for clinical use <span id="id3">[<a class="reference internal" href="#id42" title="Robert N Butler, Richard Sprott, Huber Warner, Jeffrey Bland, Richie Feuers, Michael Forster, Howard Fillit, S Mitchell Harman, Michael Hewitt, Mark Hyman, and others. Aging: the reality: biomarkers of aging: from primitive organisms to humans. The Journals of Gerontology Series A: Biological Sciences and Medical Sciences, 59(6):B560–B567, 2004. URL: https://doi.org/10.1093/gerona/59.6.b560.">Butler <em>et al.</em>, 2004</a>]</span>:</p>
<ul class="simple">
<li><p>They change with age;</p></li>
<li><p>They predict mortality better than chronological age;</p></li>
<li><p>They predict the early stages of age-related diseases;</p></li>
<li><p>They are minimally invasive and do not require painful or serious procedures.</p></li>
</ul>
<p>In addition to these characteristics, Alexey Moskalev <span id="id4">[<a class="reference internal" href="#id43" title="Alexey Moskalev. Biomarkers of human aging. Volume 10. Springer, 2019. URL: https://link.springer.com/book/10.1007/978-3-030-24970-0.">Moskalev, 2019</a>]</span> also adds that ideal aging biomarkers should be:</p>
<ul class="simple">
<li><p>Sensitive to early stages of aging;</p></li>
<li><p>Have predictability with collecting in the foreseeable time range;</p></li>
<li><p>Robust.</p></li>
</ul>
<p>Over the years, dozens of biological clocks have been published. Here are the main types.</p>
<section id="epigenetics-clocks">
<h3><a class="toc-backref" href="#id115"><span class="section-number">4.2.1. </span>Epigenetics Clocks</a><a class="headerlink" href="#epigenetics-clocks" title="Permalink to this headline">#</a></h3>
<p>During transcription, RNA is produced based on the DNA sequence, which usually leads to protein synthesis. Different proteins are required under different conditions, so there are different ways to control transcription. One of these methods is DNA methylation, in which a methyl group is attached to a nucleotide, which typically leads to suppression of transcription.</p>
<p>In mammals, methylation usually occurs in special DNA regions consisting of “CG” nucleotides, called <strong>CpG sites</strong>. The average level of cytosine methylation in these regions can be experimentally measured as a number between 0 and 1, where 1 means that all cytosines in the site are methylated on both alleles and 0 means that all cytosines are unmethylated.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p><strong>CpG site</strong> – a cytosine nucleotide followed by a guanine nucleotide, forming the pattern “CG”. In mammals, 70-80% of cytosines in these regions are methylated, and conversely, a large portion of methylated nucleotides lie in these regions.</p></li>
<li><p><strong>CpG island</strong> – a region of DNA with a high number of CpG sites.</p></li>
</ul>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Most biological clocks work on bulk data, where only the average level of methylation is known, but single-cell methods are now available that can measure the methylation status of each nucleotide <span id="id5">[<a class="reference internal" href="#id99" title="Alexandre Trapp, Csaba Kerepesi, and Vadim N Gladyshev. Profiling epigenetic age in single cells. Nature Aging, 1(12):1189–1201, 2021. URL: https://doi.org/10.1038/s43587-021-00134-3.">Trapp <em>et al.</em>, 2021</a>]</span>.</p>
</div>
<p>As one ages, methylation changes in a complex, partly predictable way <span id="id6">[<a class="reference internal" href="#id70" title="Domenico Palumbo, Ornella Affinito, Antonella Monticelli, and Sergio Cocozza. Dna methylation variability among individuals is related to cpgs cluster density and evolutionary signatures. BMC genomics, 19(1):1–9, 2018. URL: https://doi.org/10.1186/s12864-018-4618-9.">Palumbo <em>et al.</em>, 2018</a>]</span>. This means that the level of methylation can be used to predict a person’s age. Many clocks have been built on this idea. Let’s consider the most notable works.</p>
<section id="epigenetic-predictor-of-age">
<h4><a class="toc-backref" href="#id116"><span class="section-number">4.2.1.1. </span>2011 Epigenetic predictor of age</a><a class="headerlink" href="#epigenetic-predictor-of-age" title="Permalink to this headline">#</a></h4>
<p>In <span id="id7">[<a class="reference internal" href="#id44" title="Sven Bocklandt, Wen Lin, Mary E Sehl, Francisco J Sánchez, Janet S Sinsheimer, Steve Horvath, and Eric Vilain. Epigenetic predictor of age. PloS one, 6(6):e14821, 2011. URL: https://doi.org/10.1371/journal.pone.0014821.">Bocklandt <em>et al.</em>, 2011</a>]</span>, the authors built the first epigenetic clock using saliva.</p>
</section>
<section id="genome-wide-methylation-profiles-reveal-quantitative-views-of-human-aging-rates">
<h4><a class="toc-backref" href="#id117"><span class="section-number">4.2.1.2. </span>2013 Genome-wide methylation profiles reveal quantitative views of human aging rates</a><a class="headerlink" href="#genome-wide-methylation-profiles-reveal-quantitative-views-of-human-aging-rates" title="Permalink to this headline">#</a></h4>
<p>In <span id="id8">[<a class="reference internal" href="#id45" title="Gregory Hannum, Justin Guinney, Ling Zhao, LI Zhang, Guy Hughes, SriniVas Sadda, Brandy Klotzle, Marina Bibikova, Jian-Bing Fan, Yuan Gao, and others. Genome-wide methylation profiles reveal quantitative views of human aging rates. Molecular cell, 49(2):359–367, 2013. URL: https://doi.org/10.1016/j.molcel.2012.10.016.">Hannum <em>et al.</em>, 2013</a>]</span>, the clock was built on blood cells. The authors analyzed the results and found that:</p>
<ul class="simple">
<li><p>Men age 4% faster than women.</p></li>
<li><p>Clocks built on blood cells also work on other tissues, such as lung or kidney cells.</p></li>
<li><p>The authors confirmed the existence of <strong>epigenetic drift</strong>. The methylation of elderly people differs from each other more strongly than among young people. This is due to the accumulation of random errors.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p><strong>Epigenetic drift</strong> – An increase in methylation diversity with age: elderly people differ more from each other than young people. This is due to the accumulation of random errors.</p></li>
</ul>
</div>
<figure class="align-default" id="id109">
<img alt="../_images/2013paper.jpeg" src="../_images/2013paper.jpeg" />
<figcaption>
<p><span class="caption-number">Fig. 4.1 </span><span class="caption-text"><span id="id9">[<a class="reference internal" href="#id45" title="Gregory Hannum, Justin Guinney, Ling Zhao, LI Zhang, Guy Hughes, SriniVas Sadda, Brandy Klotzle, Marina Bibikova, Jian-Bing Fan, Yuan Gao, and others. Genome-wide methylation profiles reveal quantitative views of human aging rates. Molecular cell, 49(2):359–367, 2013. URL: https://doi.org/10.1016/j.molcel.2012.10.016.">Hannum <em>et al.</em>, 2013</a>]</span>. Predictions for different tissues.</span><a class="headerlink" href="#id109" title="Permalink to this image">#</a></p>
</figcaption>
</figure>
</section>
<section id="dna-methylation-age-of-human-tissues-and-cell-types">
<h4><a class="toc-backref" href="#id118"><span class="section-number">4.2.1.3. </span>2013 DNA methylation age of human tissues and cell types</a><a class="headerlink" href="#dna-methylation-age-of-human-tissues-and-cell-types" title="Permalink to this headline">#</a></h4>
<p>In <span id="id10">[<a class="reference internal" href="#id46" title="Steve Horvath. Dna methylation age of human tissues and cell types. Genome biology, 14(10):1–20, 2013. URL: https://doi.org/10.1186/gb-2013-14-10-r115.">Horvath, 2013</a>]</span>, the author:</p>
<ul class="simple">
<li><p>Built clocks using different human tissues.</p></li>
<li><p>Showed that progerias (genetic diseases characterized by accelerated aging) increase biological age.</p></li>
<li><p>Showed that reprogramming cells into <a class="reference external" href="https://en.wikipedia.org/wiki/Induced_pluripotent_stem_cell">iPSCs</a> resets biological age;</p></li>
<li><p>Showed that predictions of the clock in chimpanzee tissues correlate with chronological age;</p></li>
<li><p>Showed that cells that have undergone more divisions have a higher biological age.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p><strong>Progeria</strong> – a general term for rare genetic diseases characterized by accelerated aging.</p></li>
</ul>
</div>
</section>
<section id="an-epigenetic-biomarker-of-aging-for-lifespan-and-healthspan-phenoage">
<h4><a class="toc-backref" href="#id119"><span class="section-number">4.2.1.4. </span>2018 An epigenetic biomarker of aging for lifespan and healthspan (PhenoAge)</a><a class="headerlink" href="#an-epigenetic-biomarker-of-aging-for-lifespan-and-healthspan-phenoage" title="Permalink to this headline">#</a></h4>
<p>Ordinary biological clocks are trained to predict chronological age, expecting the model error to explain the mortality difference. But where does this assumption come from? If we could train a perfect oracle model that always correctly predicted chronological age, such a model would be useless. That is why we could predict how many years a person has left to live instead of chronological age.</p>
<p>In <span id="id11">[<a class="reference internal" href="#id50" title="Morgan E Levine, Ake T Lu, Austin Quach, Brian H Chen, Themistocles L Assimes, Stefania Bandinelli, Lifang Hou, Andrea A Baccarelli, James D Stewart, Yun Li, and others. An epigenetic biomarker of aging for lifespan and healthspan. Aging (Albany NY), 10(4):573, 2018. URL: https://doi.org/10.18632/aging.101414.">Levine <em>et al.</em>, 2018</a>]</span> the authors first trained a <a class="reference internal" href="../stat/hazard_models.html"><span class="doc std std-doc">Cox proportional regression model</span></a> to predict mortality using clinical biomarkers and age from the NHANES database. With it, they made a prediction for a dataset with blood methylation, and then trained the model to directly predict mortality from methylation data.</p>
<p>Acceleration of such a phenotypic epigenetic age is positively associated with mortality – an important property that was not present in previous clocks, so such clocks are called second generation clocks.</p>
</section>
<section id="dna-methylation-grimage-strongly-predicts-lifespan-and-healthspan">
<h4><a class="toc-backref" href="#id120"><span class="section-number">4.2.1.5. </span>2019 DNA methylation GrimAge strongly predicts lifespan and healthspan</a><a class="headerlink" href="#dna-methylation-grimage-strongly-predicts-lifespan-and-healthspan" title="Permalink to this headline">#</a></h4>
<p>The GrimAge<span id="id12">[<a class="reference internal" href="#id48" title="Ake T Lu, Austin Quach, James G Wilson, Alex P Reiner, Abraham Aviv, Kenneth Raj, Lifang Hou, Andrea A Baccarelli, Yun Li, James D Stewart, and others. Dna methylation grimage strongly predicts lifespan and healthspan. Aging (Albany NY), 11(2):303, 2019. URL: https://doi.org/10.18632/aging.101684.">Lu <em>et al.</em>, 2019</a>]</span> clock was constructed in a similar manner. As proxies for predicting mortality, the authors took the concentration of blood proteins and the number of smoked cigarette packs, which were predicted from methylation. As a result, a model was obtained that better predicted the date of death and the onset of age-related diseases than previous models.</p>
<div class="tip admonition">
<p class="admonition-title">Oracle Model</p>
<p>In <span id="id13">[<a class="reference internal" href="#id49" title="Qian Zhang, Costanza L Vallerga, Rosie M Walker, Tian Lin, Anjali K Henders, Grant W Montgomery, Ji He, Dongsheng Fan, Javed Fowdar, Martin Kennedy, and others. Improved precision of epigenetic clock estimates across tissues and its implication for biological ageing. Genome medicine, 11(1):1–11, 2019. URL: https://doi.org/10.1186/s13073-019-0667-1.">Zhang <em>et al.</em>, 2019</a>]</span> the authors trained the most accurate epigenetic clocks with an RSME error of 2.04 years. While the model performed better than all previous ones, its acceleration of biological age no longer predicted mortality and was no longer associated with diseases, rendering clinical use of such clocks meaningless.</p>
</div>
</section>
</section>
<section id="other-clocks">
<h3><a class="toc-backref" href="#id121"><span class="section-number">4.2.2. </span>Other Clocks</a><a class="headerlink" href="#other-clocks" title="Permalink to this headline">#</a></h3>
<section id="transcriptomic-clocks">
<h4><a class="toc-backref" href="#id122"><span class="section-number">4.2.2.1. </span>Transcriptomic Clocks</a><a class="headerlink" href="#transcriptomic-clocks" title="Permalink to this headline">#</a></h4>
<p>DNA transcription is the process by which mRNA molecules, which are used to create proteins, are synthesized. By analyzing the full transcriptome, or all of the RNA molecules, of a cell, we can sequence and compare the transcriptomes of healthy individuals with those of sick individuals. This helps to uncover the mechanisms behind diseases and potentially discover new treatments<span id="id14">[<a class="reference internal" href="#id57" title="Xiaonan Yang, Ling Kui, Min Tang, Dawei Li, Kunhua Wei, Wei Chen, Jianhua Miao, and Yang Dong. High-throughput transcriptome profiling in drug and biomarker discovery. Frontiers in genetics, 11:19, 2020. URL: https://doi.org/10.3389/fgene.2020.00019.">Yang <em>et al.</em>, 2020</a>]</span>.</p>
<p>We could learn more about aging by comparing the transcriptomes of old people with those of young people. One way to do this is to build aging clocks on the transcriptome and try to understand why the model turned out to be that way. This is what was done in <span id="id15">[<a class="reference internal" href="#id52" title="Polina Mamoshina, Marina Volosnikova, Ivan V Ozerov, Evgeny Putin, Ekaterina Skibina, Franco Cortese, and Alex Zhavoronkov. Machine learning on human muscle transcriptomic data for biomarker discovery and tissue-specific drug target identification. Frontiers in genetics, 9:242, 2018. URL: https://doi.org/10.3389/fgene.2018.00242.">Mamoshina <em>et al.</em>, 2018</a>]</span>, where the authors trained clocks on the transcriptomes of human muscles. The clocks were less accurate (MAE 6.19 years) than methylome-based clocks, but feature importance techniques proposed new targets for potential geroprotectors.</p>
<p>In addition to interpretability, transcriptome-based clocks have another advantage. C. elegans roundworms do not have 5mC nucleotides, so epigenetic clocks cannot be built for them in the traditional sense. Therefore, transcriptome-based clocks were built for c. elegans <span id="id16">[<a class="reference internal" href="#id51" title="David H Meyer and Björn Schumacher. Bit age: a transcriptome-based aging clock near the theoretical limit of accuracy. Aging cell, 20(3):e13320, 2021. URL: https://doi.org/10.1111/acel.13320.">Meyer and Schumacher, 2021</a>, <a class="reference internal" href="#id100" title="Andrei E Tarkhov, Ramani Alla, Srinivas Ayyadevara, Mikhail Pyatnitskiy, Leonid I Menshikov, Robert J Shmookler Reis, and Peter O Fedichev. A universal transcriptomic signature of age reveals the temporal scaling of caenorhabditis elegans aging trajectories. Scientific reports, 9(1):1–18, 2019. URL: https://doi.org/10.1038/s41598-019-43075-z.">Tarkhov <em>et al.</em>, 2019</a>]</span> that work for worms of different strains living in different conditions.</p>
</section>
<section id="blood-test-clocks">
<h4><a class="toc-backref" href="#id123"><span class="section-number">4.2.2.2. </span>Blood Test Clocks</a><a class="headerlink" href="#blood-test-clocks" title="Permalink to this headline">#</a></h4>
<p>Transcriptome and methylome are obtained through expensive sequencing. It would be more convenient for clinical practice to build an aging clock on cheap complete blood count. This was done in <span id="id17">[<a class="reference internal" href="#id55" title="Evgeny Putin, Polina Mamoshina, Alexander Aliper, Mikhail Korzinkin, Alexey Moskalev, Alexey Kolosov, Alexander Ostrovskiy, Charles Cantor, Jan Vijg, and Alex Zhavoronkov. Deep biomarkers of human aging: application of deep neural networks to biomarker development. Aging (Albany NY), 8(5):1021, 2016. URL: https://doi.org/10.18632/aging.100968.">Putin <em>et al.</em>, 2016</a>]</span>, and the feature importance technique showed that the most important predictors of age are albumin, glucose, alkaline phosphatase, urea, and erythrocytes. The model can be conveniently used through the website <a class="reference external" href="http://aging.ai">http://aging.ai</a>.</p>
<p>In the following article <span id="id18">[<a class="reference internal" href="#id54" title="Polina Mamoshina, Kirill Kochetov, Evgeny Putin, Franco Cortese, Alexander Aliper, Won-Suk Lee, Sung-Min Ahn, Lee Uhn, Neil Skjodt, Olga Kovalchuk, and others. Population specific biomarkers of human aging: a big data study using south korean, canadian, and eastern european patient populations. The Journals of Gerontology: Series A, 73(11):1482–1490, 2018. URL: https://doi.org/10.1093/gerona/gly005.">Mamoshina <em>et al.</em>, 2018</a>]</span>, the authors improved the accuracy of the model and reduced the number of necessary measurements. The authors found that the quality of the model varies between patients from different countries: Canada, Eastern Europe, and South Korea.</p>
</section>
</section>
</section>
<section id="application-of-biological-clocks">
<h2><a class="toc-backref" href="#id124"><span class="section-number">4.3. </span>Application of Biological Clocks</a><a class="headerlink" href="#application-of-biological-clocks" title="Permalink to this headline">#</a></h2>
<p>Acceleration of biological age meets the <a class="reference internal" href="#bioclocks-type"><span class="std std-ref">criteria for a marker of aging</span></a>. For example, it predicts mortality from various causes<span id="id19">[<a class="reference internal" href="#id58" title="Riccardo E Marioni, Sonia Shah, Allan F McRae, Brian H Chen, Elena Colicino, Sarah E Harris, Jude Gibson, Anjali K Henders, Paul Redmond, Simon R Cox, and others. Dna methylation age of blood predicts all-cause mortality in later life. Genome biology, 16(1):1–12, 2015. URL: https://doi.org/10.1186/s13059-015-0584-6.">Marioni <em>et al.</em>, 2015</a>]</span>. An increase in epigenetic clock of the first generation is not associated with a deterioration of physical and cognitive functions<span id="id20">[<a class="reference internal" href="#id59" title="Riccardo E Marioni, Sonia Shah, Allan F McRae, Stuart J Ritchie, Graciela Muniz-Terrera, Sarah E Harris, Jude Gibson, Paul Redmond, Simon R Cox, Alison Pattie, and others. The epigenetic clock is correlated with physical and cognitive fitness in the lothian birth cohort 1936. International journal of epidemiology, 44(4):1388–1396, 2015. URL: https://doi.org/10.1093/ije/dyu277.">Marioni <em>et al.</em>, 2015</a>]</span>, but is associated for clocks of the second generation<span id="id21">[<a class="reference internal" href="#id64" title="Jane Maddock, Juan Castillo-Fernandez, Andrew Wong, Rachel Cooper, Marcus Richards, Ken K Ong, George B Ploubidis, Alissa Goodman, Diana Kuh, Jordana T Bell, and others. Dna methylation age and physical and cognitive aging. The Journals of Gerontology: Series A, 75(3):504–511, 2020. URL: https://doi.org/10.1093/gerona/glz246.">Maddock <em>et al.</em>, 2020</a>]</span>.</p>
<p>Obesity <span id="id22">[<a class="reference internal" href="#id60" title="Steve Horvath, Wiebke Erhart, Mario Brosch, Ole Ammerpohl, Witigo von Schönfels, Markus Ahrens, Nils Heits, Jordana T Bell, Pei-Chien Tsai, Tim D Spector, and others. Obesity accelerates epigenetic aging of human liver. Proceedings of the National Academy of Sciences, 111(43):15538–15543, 2014. URL: https://doi.org/10.1073/pnas.1412759111.">Horvath <em>et al.</em>, 2014</a>]</span> and Down syndrome <span id="id23">[<a class="reference internal" href="#id61" title="Steve Horvath, Paolo Garagnani, Maria Giulia Bacalini, Chiara Pirazzini, Stefano Salvioli, Davide Gentilini, Anna Maria Di Blasio, Cristina Giuliani, Spencer Tung, Harry V Vinters, and others. Accelerated epigenetic aging in down syndrome. Aging cell, 14(3):491–495, 2015. URL: https://doi.org/10.1111/acel.12325.">Horvath <em>et al.</em>, 2015</a>]</span> increase biological age. HIV also increases biological age even in brain tissue, although it is not clear why <span id="id24">[<a class="reference internal" href="#id62" title="Steve Horvath and Andrew J Levine. Hiv-1 infection accelerates age according to the epigenetic clock. The Journal of infectious diseases, 212(10):1563–1573, 2015. URL: https://doi.org/10.1093/infdis/jiv277.">Horvath and Levine, 2015</a>]</span>.</p>
<p>In 2006, the CMAP dataset was published <span id="id25">[<a class="reference internal" href="#id65" title="Justin Lamb, Emily D Crawford, David Peck, Joshua W Modell, Irene C Blat, Matthew J Wrobel, Jim Lerner, Jean-Philippe Brunet, Aravind Subramanian, Kenneth N Ross, and others. The connectivity map: using gene-expression signatures to connect small molecules, genes, and disease. science, 313(5795):1929–1935, 2006. URL: https://doi.org/10.1126/science.1132939.">Lamb <em>et al.</em>, 2006</a>]</span>, in which the authors measured how the transcriptome of cells changes after exposure to 1300 different drugs. Using this dataset and biological clocks, it is possible to find new drugs. For this, in <span id="id26">[<a class="reference internal" href="#id41" title="Georges E Janssens, Xin-Xuan Lin, Lluís Millan-Ariño, Alan Kavšek, Ilke Sen, Renée I Seinstra, Nicholas Stroustrup, Ellen AA Nollen, and Christian G Riedel. Transcriptomics-based screening identifies pharmacological inhibition of hsp90 as a means to defer aging. Cell reports, 27(2):467–480, 2019. URL: https://doi.org/10.1016/j.celrep.2019.03.044.">Janssens <em>et al.</em>, 2019</a>]</span> the authors trained transcriptomic clocks on 51 human tissue and then searched in CMAP for drugs that, according to the model, reduce biological age. As a result, known geroprotectors and several new molecules were found, which were tested on worms c. elegans. The experiment confirmed that inhibitors of the protein Hsp90 extend life.</p>
<section id="open-questions">
<h3><a class="toc-backref" href="#id125"><span class="section-number">4.3.1. </span>Open Questions</a><a class="headerlink" href="#open-questions" title="Permalink to this headline">#</a></h3>
<p>During the search for drugs that lower the predicted model age, we find geroprotectors that extend life. Does this mean a causal relationship? Can we call a molecule that reduces biological age a geroprotector, but for which a costly, long-term clinical trial has not yet been conducted that proves a reduction in mortality? There are several problems <span id="id27">[<a class="reference internal" href="#id71" title="Christopher G Bell, Robert Lowe, Peter D Adams, Andrea A Baccarelli, Stephan Beck, Jordana T Bell, Brock C Christensen, Vadim N Gladyshev, Bastiaan T Heijmans, Steve Horvath, and others. Dna methylation aging clocks: challenges and recommendations. Genome biology, 20(1):1–24, 2019. URL: https://doi.org/10.1186/s13059-019-1824-y.">Bell <em>et al.</em>, 2019</a>, <a class="reference internal" href="#id66" title="Nicholas J Schork, Brett Beaulieu-Jones, Winnie Liang, Susan Smalley, and Laura H Goetz. Does modulation of an epigenetic clock define a geroprotector? Advances in geriatric medicine and research, 2022. URL: https://doi.org/10.20900%2Fagmr20220002.">Schork <em>et al.</em>, 2022</a>]</span>.</p>
<p><strong>Predictions of different clocks are weakly correlated with each other. Epigenetic clocks are trained on different CpG sites.</strong></p>
<figure class="align-default" id="id110">
<img alt="../_images/different_cpgs.jpeg" src="../_images/different_cpgs.jpeg" />
<figcaption>
<p><span class="caption-number">Fig. 4.2 </span><span class="caption-text"><span id="id28">[<a class="reference internal" href="#id101" title="Fedor Galkin, Polina Mamoshina, Alex Aliper, João Pedro de Magalhães, Vadim N Gladyshev, and Alex Zhavoronkov. Biohorology and biomarkers of aging: current state-of-the-art, challenges and opportunities. Ageing Research Reviews, 60:101050, 2020. URL: https://doi.org/10.1016/j.arr.2020.101050.">Galkin <em>et al.</em>, 2020</a>]</span></span><a class="headerlink" href="#id110" title="Permalink to this image">#</a></p>
</figcaption>
</figure>
<p>Does this mean that clocks measure different aspects of aging? Or that clock predictions are very noisy? The authors of <span id="id29">[<a class="reference internal" href="#id67" title="Zuyun Liu, Diana Leung, Kyra Thrush, Wei Zhao, Scott Ratliff, Toshiko Tanaka, Lauren L Schmitz, Jennifer A Smith, Luigi Ferrucci, and Morgan E Levine. Underlying features of epigenetic aging clocks in vivo and in vitro. Aging cell, 19(10):e13229, 2020. URL: https://doi.org/10.1111/acel.13229.">Liu <em>et al.</em>, 2020</a>]</span> show that clocks capture different aspects of aging. If this is the case, it would be important to find these separate aspects and make separate clocks for them.</p>
<p><strong>Predictions of aging clocks poorly correlate with other classic biomarkers, such as blood pressure or glucose levels.</strong>
This may be because aging clocks only measure one aspect of aging, or because aging clocks measure something more fundamental than what other clinical biomarkers measure. Could molecules that reduce biological age but do not affect classic biomarkers extend lifespan?</p>
<p><strong>The cause-and-effect relationship is unclear.</strong>
To what extent are epigenetic changes the cause of aging, and to what extent are they a consequence?
In the first generation of clock models, training was done on predicting chronological age. This means that the data on elderly people in the dataset are from those who have managed to live to old age. Could this distort the model?</p>
<p>The authors of <span id="id30">[<a class="reference internal" href="#id69" title="Paul G Nelson, Daniel EL Promislow, and Joanna Masel. Biomarkers for aging identified in cross-sectional studies tend to be non-causative. The Journals of Gerontology: Series A, 75(3):466–472, 2020. URL: https://doi.org/10.1093/gerona/glz174.">Nelson <em>et al.</em>, 2020</a>]</span> show that it can. The authors simulated random changes in methylation and assumed two things: mortality increases with the number of changes in CpG sites and changes in different CpG sites increase mortality differently. They trained an Elasticnet model on synthetic data and showed that as a result of L1 regularization, it selects non-causative sites that contribute little to mortality</p>
<div class="tip admonition">
<p class="admonition-title">F.A.Q.</p>
<p><strong>Q:</strong> Is it true that epigenetic clocks simply show differences in tissue composition?</p>
<p><strong>A:</strong> It depends on the clock. If it is a multi-tissue <span id="id31">[<a class="reference internal" href="#id46" title="Steve Horvath. Dna methylation age of human tissues and cell types. Genome biology, 14(10):1–20, 2013. URL: https://doi.org/10.1186/gb-2013-14-10-r115.">Horvath, 2013</a>]</span>, then no: you can train the clock on one tissue and it will work on another. If it is a second generation <span id="id32">[<a class="reference internal" href="#id50" title="Morgan E Levine, Ake T Lu, Austin Quach, Brian H Chen, Themistocles L Assimes, Stefania Bandinelli, Lifang Hou, Andrea A Baccarelli, James D Stewart, Yun Li, and others. An epigenetic biomarker of aging for lifespan and healthspan. Aging (Albany NY), 10(4):573, 2018. URL: https://doi.org/10.18632/aging.101414.">Levine <em>et al.</em>, 2018</a>, <a class="reference internal" href="#id48" title="Ake T Lu, Austin Quach, James G Wilson, Alex P Reiner, Abraham Aviv, Kenneth Raj, Lifang Hou, Andrea A Baccarelli, Yun Li, James D Stewart, and others. Dna methylation grimage strongly predicts lifespan and healthspan. Aging (Albany NY), 11(2):303, 2019. URL: https://doi.org/10.18632/aging.101684.">Lu <em>et al.</em>, 2019</a>]</span> phenotypic clock or a single-tissue clock, then it is possible.</p>
<p><strong>Q:</strong> Could it be that epigenetic clocks work only because they distinguish senescent cells?</p>
<p><strong>A:</strong> It does not seem so, because they work in immortal non-senescent cells, such as immortalized B cells or in embryonic cells. Cell aging and senescence are associated, but they are not the same. See the discussion and experiments in <span id="id33">[<a class="reference internal" href="#id63" title="Donna Lowe, Steve Horvath, and Kenneth Raj. Epigenetic clock analyses of cellular senescence and ageing. Oncotarget, 7(8):8524–8531, 2016. URL: https://doi.org/10.18632/oncotarget.7383.">Lowe <em>et al.</em>, 2016</a>]</span>.</p>
<p><strong>Q:</strong> Could it be that epigenetic clocks only measure the number of cell divisions?</p>
<p><strong>A:</strong> It does not seem so, because they work in non-proliferating cells, such as brain cells.</p>
</div>
</section>
</section>
<section id="linear-regression">
<h2><a class="toc-backref" href="#id126"><span class="section-number">4.4. </span>Linear Regression</a><a class="headerlink" href="#linear-regression" title="Permalink to this headline">#</a></h2>
<p>As we can see, developing biological clocks is an important task in aging biology. Most biological clock models predict either chronological age or risk of death based on a person’s parameters. But how does this technically happen? How can we make our own biological clocks for our own data?</p>
<p>Almost all biological clocks are linear regression. Linear regression is ubiquitous, so we assume that the reader is already familiar with it, so we will only remind you of the main ideas. For a first introduction to linear regression, we recommend practical tutorials<a class="reference external" href="https://scikit-learn.org/stable/modules/linear_model.html">[1]</a>, <a class="reference external" href="https://www.kaggle.com/learn/intro-to-machine-learning">[2]</a>, <a class="reference external" href="https://d2l.ai/chapter_linear-regression/index.html">[3]</a>.</p>
<p>In the linear regression model, we assume that the predicted value is linearly dependent on the input data. Let <span class="math notranslate nohighlight">\(y\)</span> be the chronological age that we are trying to predict. <span class="math notranslate nohighlight">\(x_{1}, x_{2}, \dots x_{n}\)</span> are known input parameters, such as blood test results. Then in the linear regression model, we approximate <span class="math notranslate nohighlight">\(y\)</span> as a linear function:</p>
<div class="math notranslate nohighlight">
\[
a_1 x_1 + a_2 x_2 + \dots + a_n x_n = y + \epsilon
\]</div>
<p>where <span class="math notranslate nohighlight">\(a_1, a_2, \dots a_n\)</span> are coefficients that we need to select, and <span class="math notranslate nohighlight">\(\epsilon\)</span> represents normal noise, explaining, for example, that blood tests are measured with error:</p>
<div class="math notranslate nohighlight">
\[
\epsilon \sim \mathcal{N}(0,\,\sigma^{2})
\]</div>
<p>We want to select such coefficients <span class="math notranslate nohighlight">\(a_1, a_2, \dots a_n\)</span> so that the model predicts chronological age as accurately as possible. In other words, we want to minimize the error:</p>
<div class="math notranslate nohighlight">
\[
min_{a_1,\dots a_n} | (a_1 x_1 + a_2 x_2 + \dots + a_n x_n) - y |
\]</div>
<p>In practice, the error is minimized using gradient descent, where at each iteration the coefficients <span class="math notranslate nohighlight">\(a_1, a_2, \dots a_n\)</span> are shifted in such a way as to make the error as small as possible.</p>
<section id="overfitting-and-regularization">
<h3><a class="toc-backref" href="#id127"><span class="section-number">4.4.1. </span>Overfitting and Regularization</a><a class="headerlink" href="#overfitting-and-regularization" title="Permalink to this headline">#</a></h3>
<p>The more parameters a model has, and the less data in the training dataset, the easier it is for the model to simply memorize the entire training dataset. In this case, the linear function approximating chronological age will poorly describe the real world, and we will see a large error on the test dataset. This is called overfitting and can easily occur if you have a small dataset and a large number of parameters.</p>
<p>To deal with overfitting, regularization can be applied – that is, imposing some constraints to make the model simpler. For example, if you overfit a linear regression model, you will notice that its coefficients become very large:</p>
<div class="math notranslate nohighlight">
\[
1863526.326 x_1 - 7532253.1663 x_2 + \dots + 52.1 x_n = \hat{y}
\]</div>
<p>It is unlikely that chronological age depends on blood test analyses with such coefficients in real life, so you can try penalizing the model for having too large coefficients during training. To do this, during gradient descent we will minimize both the error and the sum of the absolute values of the coefficients.</p>
<div class="math notranslate nohighlight">
\[
min_{a_1,\dots a_n} | (a_1 x_1 + a_2 x_2 + \dots + a_n x_n) - y | + \lambda \sum_{i} |a_i|
\]</div>
<p>where <span class="math notranslate nohighlight">\(\lambda\)</span> is a coefficient representing the strength of regularization. This method of regularization is called L1 regularization and can improve the model’s performance on the test dataset in practice.</p>
</section>
<section id="l1-and-l2-regularization">
<h3><a class="toc-backref" href="#id128"><span class="section-number">4.4.2. </span>L1 and L2 regularization</a><a class="headerlink" href="#l1-and-l2-regularization" title="Permalink to this headline">#</a></h3>
<p>A model can be regularized in different ways, but the two most common methods are L1 and L2 regularization. L2 regularization differs in that the error is not just the absolute value of the coefficients, but also their squares added to the error:”</p>
<div class="math notranslate nohighlight">
\[
min_{a_1,\dots a_n} | (a_1 x_1 + a_2 x_2 + \dots + a_n x_n) - y | + \lambda \sum_{i} a_{i}^{2}
\]</div>
<p>Both regularization methods help to find small coefficients, but the difference between them is that L2 regularization more heavily penalizes large coefficients and less heavily penalizes small coefficients. For example, if the model has two coefficients of 0.5 and 2, then with L2 regularization their contribution to the penalty becomes 0.25 and 4.</p>
<p>Both methods help to combat overfitting, but L1 is also useful in that the coefficients of unimportant features can quickly become zero, which gives us a useful signal that this blood parameter does not help the model predict chronological age. In practice, the regularization that helps improve the model’s performance on the test dataset is used.</p>
<figure class="align-default" id="id111">
<img alt="../_images/ridge.png" src="../_images/ridge.png" />
<figcaption>
<p><span class="caption-number">Fig. 4.3 </span><span class="caption-text">Effect of L1-regularization on the coefficients. The more penalty coefficient (alpha in picture), the less the coefficients. From <a class="reference external" href="https://scikit-learn.org/stable/auto_examples/linear_model/plot_ridge_path.html">sklearn tutorial</a>.</span><a class="headerlink" href="#id111" title="Permalink to this image">#</a></p>
</figcaption>
</figure>
</section>
<section id="elasticnet">
<h3><a class="toc-backref" href="#id129"><span class="section-number">4.4.3. </span>Elasticnet</a><a class="headerlink" href="#elasticnet" title="Permalink to this headline">#</a></h3>
<p>You can choose not to choose between the two regularization methods and simply apply both. Sometimes this works better than one regularization alone:</p>
<div class="math notranslate nohighlight">
\[
min_{a_1,\dots a_n} | (a_1 x_1 + a_2 x_2 + \dots + a_n x_n) - y | + \lambda_1 \sum_{i} a_{i}^{2} + \lambda_2 \sum_{i} |a_{i}|
\]</div>
<p>Such a model is called ElasticNet and is used in the construction of several clocks.</p>
</section>
</section>
<section id="conclusion">
<h2><a class="toc-backref" href="#id130"><span class="section-number">4.5. </span>Conclusion</a><a class="headerlink" href="#conclusion" title="Permalink to this headline">#</a></h2>
<p>Aging clocks are an important area of research that has the potential to greatly improve our understanding of the aging process and lead to the development of interventions to promote healthy aging. These clocks are biological markers that can be used to measure and predict the biological age of an individual. They can be based on a variety of factors, such as epigenetic modifications, blood tests, and gene expression patterns. While there has been significant progress in understanding and utilizing aging clocks, there are still many open questions and areas for further research.</p>
<p>One important aspect of aging clock research is the use of statistical analysis to make these clocks. Linear regression is a commonly used statistical method in this context, as it allows researchers to identify correlations and trends in the data. However, it is important to recognize that basic linear regression has limitations and additional regularizations may be necessary in certain circumstances.</p>
<p>Overall, the study of aging clocks has the potential to provide insights into the underlying mechanisms of aging and may lead to the development of interventions that can promote healthy aging and extend the human lifespan. While there is still much to learn about aging clocks and the aging process, the potential benefits of this research make it an important area of study for the scientific community.</p>
</section>
<section id="credits">
<h2><a class="toc-backref" href="#id131"><span class="section-number">4.6. </span>Credits</a><a class="headerlink" href="#credits" title="Permalink to this headline">#</a></h2>
<p>This text prepared by <a class="reference external" href="https://www.linkedin.com/in/simon-steshin-506ab1197/">Simon Steshin</a>.</p>
<div class="docutils container" id="id34">
<dl class="citation">
<dt class="label" id="id40"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p>Nir Barzilai, Jill P Crandall, Stephen B Kritchevsky, and Mark A Espeland. Metformin as a tool to target aging. <em>Cell metabolism</em>, 23(6):1060–1065, 2016. URL: <a class="reference external" href="https://doi.org/10.1016/j.cmet.2016.05.011">https://doi.org/10.1016/j.cmet.2016.05.011</a>.</p>
</dd>
<dt class="label" id="id71"><span class="brackets"><a class="fn-backref" href="#id27">2</a></span></dt>
<dd><p>Christopher G Bell, Robert Lowe, Peter D Adams, Andrea A Baccarelli, Stephan Beck, Jordana T Bell, Brock C Christensen, Vadim N Gladyshev, Bastiaan T Heijmans, Steve Horvath, and others. Dna methylation aging clocks: challenges and recommendations. <em>Genome biology</em>, 20(1):1–24, 2019. URL: <a class="reference external" href="https://doi.org/10.1186/s13059-019-1824-y">https://doi.org/10.1186/s13059-019-1824-y</a>.</p>
</dd>
<dt class="label" id="id44"><span class="brackets"><a class="fn-backref" href="#id7">3</a></span></dt>
<dd><p>Sven Bocklandt, Wen Lin, Mary E Sehl, Francisco J Sánchez, Janet S Sinsheimer, Steve Horvath, and Eric Vilain. Epigenetic predictor of age. <em>PloS one</em>, 6(6):e14821, 2011. URL: <a class="reference external" href="https://doi.org/10.1371/journal.pone.0014821">https://doi.org/10.1371/journal.pone.0014821</a>.</p>
</dd>
<dt class="label" id="id42"><span class="brackets"><a class="fn-backref" href="#id3">4</a></span></dt>
<dd><p>Robert N Butler, Richard Sprott, Huber Warner, Jeffrey Bland, Richie Feuers, Michael Forster, Howard Fillit, S Mitchell Harman, Michael Hewitt, Mark Hyman, and others. Aging: the reality: biomarkers of aging: from primitive organisms to humans. <em>The Journals of Gerontology Series A: Biological Sciences and Medical Sciences</em>, 59(6):B560–B567, 2004. URL: <a class="reference external" href="https://doi.org/10.1093/gerona/59.6.b560">https://doi.org/10.1093/gerona/59.6.b560</a>.</p>
</dd>
<dt class="label" id="id101"><span class="brackets"><a class="fn-backref" href="#id28">5</a></span></dt>
<dd><p>Fedor Galkin, Polina Mamoshina, Alex Aliper, João Pedro de Magalhães, Vadim N Gladyshev, and Alex Zhavoronkov. Biohorology and biomarkers of aging: current state-of-the-art, challenges and opportunities. <em>Ageing Research Reviews</em>, 60:101050, 2020. URL: <a class="reference external" href="https://doi.org/10.1016/j.arr.2020.101050">https://doi.org/10.1016/j.arr.2020.101050</a>.</p>
</dd>
<dt class="label" id="id45"><span class="brackets">6</span><span class="fn-backref">(<a href="#id8">1</a>,<a href="#id9">2</a>)</span></dt>
<dd><p>Gregory Hannum, Justin Guinney, Ling Zhao, LI Zhang, Guy Hughes, SriniVas Sadda, Brandy Klotzle, Marina Bibikova, Jian-Bing Fan, Yuan Gao, and others. Genome-wide methylation profiles reveal quantitative views of human aging rates. <em>Molecular cell</em>, 49(2):359–367, 2013. URL: <a class="reference external" href="https://doi.org/10.1016/j.molcel.2012.10.016">https://doi.org/10.1016/j.molcel.2012.10.016</a>.</p>
</dd>
<dt class="label" id="id46"><span class="brackets">7</span><span class="fn-backref">(<a href="#id10">1</a>,<a href="#id31">2</a>)</span></dt>
<dd><p>Steve Horvath. Dna methylation age of human tissues and cell types. <em>Genome biology</em>, 14(10):1–20, 2013. URL: <a class="reference external" href="https://doi.org/10.1186/gb-2013-14-10-r115">https://doi.org/10.1186/gb-2013-14-10-r115</a>.</p>
</dd>
<dt class="label" id="id60"><span class="brackets"><a class="fn-backref" href="#id22">8</a></span></dt>
<dd><p>Steve Horvath, Wiebke Erhart, Mario Brosch, Ole Ammerpohl, Witigo von Schönfels, Markus Ahrens, Nils Heits, Jordana T Bell, Pei-Chien Tsai, Tim D Spector, and others. Obesity accelerates epigenetic aging of human liver. <em>Proceedings of the National Academy of Sciences</em>, 111(43):15538–15543, 2014. URL: <a class="reference external" href="https://doi.org/10.1073/pnas.1412759111">https://doi.org/10.1073/pnas.1412759111</a>.</p>
</dd>
<dt class="label" id="id61"><span class="brackets"><a class="fn-backref" href="#id23">9</a></span></dt>
<dd><p>Steve Horvath, Paolo Garagnani, Maria Giulia Bacalini, Chiara Pirazzini, Stefano Salvioli, Davide Gentilini, Anna Maria Di Blasio, Cristina Giuliani, Spencer Tung, Harry V Vinters, and others. Accelerated epigenetic aging in down syndrome. <em>Aging cell</em>, 14(3):491–495, 2015. URL: <a class="reference external" href="https://doi.org/10.1111/acel.12325">https://doi.org/10.1111/acel.12325</a>.</p>
</dd>
<dt class="label" id="id62"><span class="brackets"><a class="fn-backref" href="#id24">10</a></span></dt>
<dd><p>Steve Horvath and Andrew J Levine. Hiv-1 infection accelerates age according to the epigenetic clock. <em>The Journal of infectious diseases</em>, 212(10):1563–1573, 2015. URL: <a class="reference external" href="https://doi.org/10.1093/infdis/jiv277">https://doi.org/10.1093/infdis/jiv277</a>.</p>
</dd>
<dt class="label" id="id41"><span class="brackets">11</span><span class="fn-backref">(<a href="#id2">1</a>,<a href="#id26">2</a>)</span></dt>
<dd><p>Georges E Janssens, Xin-Xuan Lin, Lluís Millan-Ariño, Alan Kavšek, Ilke Sen, Renée I Seinstra, Nicholas Stroustrup, Ellen AA Nollen, and Christian G Riedel. Transcriptomics-based screening identifies pharmacological inhibition of hsp90 as a means to defer aging. <em>Cell reports</em>, 27(2):467–480, 2019. URL: <a class="reference external" href="https://doi.org/10.1016/j.celrep.2019.03.044">https://doi.org/10.1016/j.celrep.2019.03.044</a>.</p>
</dd>
<dt class="label" id="id65"><span class="brackets"><a class="fn-backref" href="#id25">12</a></span></dt>
<dd><p>Justin Lamb, Emily D Crawford, David Peck, Joshua W Modell, Irene C Blat, Matthew J Wrobel, Jim Lerner, Jean-Philippe Brunet, Aravind Subramanian, Kenneth N Ross, and others. The connectivity map: using gene-expression signatures to connect small molecules, genes, and disease. <em>science</em>, 313(5795):1929–1935, 2006. URL: <a class="reference external" href="https://doi.org/10.1126/science.1132939">https://doi.org/10.1126/science.1132939</a>.</p>
</dd>
<dt class="label" id="id50"><span class="brackets">13</span><span class="fn-backref">(<a href="#id11">1</a>,<a href="#id32">2</a>)</span></dt>
<dd><p>Morgan E Levine, Ake T Lu, Austin Quach, Brian H Chen, Themistocles L Assimes, Stefania Bandinelli, Lifang Hou, Andrea A Baccarelli, James D Stewart, Yun Li, and others. An epigenetic biomarker of aging for lifespan and healthspan. <em>Aging (Albany NY)</em>, 10(4):573, 2018. URL: <a class="reference external" href="https://doi.org/10.18632/aging.101414">https://doi.org/10.18632/aging.101414</a>.</p>
</dd>
<dt class="label" id="id67"><span class="brackets"><a class="fn-backref" href="#id29">14</a></span></dt>
<dd><p>Zuyun Liu, Diana Leung, Kyra Thrush, Wei Zhao, Scott Ratliff, Toshiko Tanaka, Lauren L Schmitz, Jennifer A Smith, Luigi Ferrucci, and Morgan E Levine. Underlying features of epigenetic aging clocks in vivo and in vitro. <em>Aging cell</em>, 19(10):e13229, 2020. URL: <a class="reference external" href="https://doi.org/10.1111/acel.13229">https://doi.org/10.1111/acel.13229</a>.</p>
</dd>
<dt class="label" id="id63"><span class="brackets"><a class="fn-backref" href="#id33">15</a></span></dt>
<dd><p>Donna Lowe, Steve Horvath, and Kenneth Raj. Epigenetic clock analyses of cellular senescence and ageing. <em>Oncotarget</em>, 7(8):8524–8531, 2016. URL: <a class="reference external" href="https://doi.org/10.18632/oncotarget.7383">https://doi.org/10.18632/oncotarget.7383</a>.</p>
</dd>
<dt class="label" id="id48"><span class="brackets">16</span><span class="fn-backref">(<a href="#id12">1</a>,<a href="#id32">2</a>)</span></dt>
<dd><p>Ake T Lu, Austin Quach, James G Wilson, Alex P Reiner, Abraham Aviv, Kenneth Raj, Lifang Hou, Andrea A Baccarelli, Yun Li, James D Stewart, and others. Dna methylation grimage strongly predicts lifespan and healthspan. <em>Aging (Albany NY)</em>, 11(2):303, 2019. URL: <a class="reference external" href="https://doi.org/10.18632/aging.101684">https://doi.org/10.18632/aging.101684</a>.</p>
</dd>
<dt class="label" id="id64"><span class="brackets"><a class="fn-backref" href="#id21">17</a></span></dt>
<dd><p>Jane Maddock, Juan Castillo-Fernandez, Andrew Wong, Rachel Cooper, Marcus Richards, Ken K Ong, George B Ploubidis, Alissa Goodman, Diana Kuh, Jordana T Bell, and others. Dna methylation age and physical and cognitive aging. <em>The Journals of Gerontology: Series A</em>, 75(3):504–511, 2020. URL: <a class="reference external" href="https://doi.org/10.1093/gerona/glz246">https://doi.org/10.1093/gerona/glz246</a>.</p>
</dd>
<dt class="label" id="id54"><span class="brackets"><a class="fn-backref" href="#id18">18</a></span></dt>
<dd><p>Polina Mamoshina, Kirill Kochetov, Evgeny Putin, Franco Cortese, Alexander Aliper, Won-Suk Lee, Sung-Min Ahn, Lee Uhn, Neil Skjodt, Olga Kovalchuk, and others. Population specific biomarkers of human aging: a big data study using south korean, canadian, and eastern european patient populations. <em>The Journals of Gerontology: Series A</em>, 73(11):1482–1490, 2018. URL: <a class="reference external" href="https://doi.org/10.1093/gerona/gly005">https://doi.org/10.1093/gerona/gly005</a>.</p>
</dd>
<dt class="label" id="id52"><span class="brackets"><a class="fn-backref" href="#id15">19</a></span></dt>
<dd><p>Polina Mamoshina, Marina Volosnikova, Ivan V Ozerov, Evgeny Putin, Ekaterina Skibina, Franco Cortese, and Alex Zhavoronkov. Machine learning on human muscle transcriptomic data for biomarker discovery and tissue-specific drug target identification. <em>Frontiers in genetics</em>, 9:242, 2018. URL: <a class="reference external" href="https://doi.org/10.3389/fgene.2018.00242">https://doi.org/10.3389/fgene.2018.00242</a>.</p>
</dd>
<dt class="label" id="id58"><span class="brackets"><a class="fn-backref" href="#id19">20</a></span></dt>
<dd><p>Riccardo E Marioni, Sonia Shah, Allan F McRae, Brian H Chen, Elena Colicino, Sarah E Harris, Jude Gibson, Anjali K Henders, Paul Redmond, Simon R Cox, and others. Dna methylation age of blood predicts all-cause mortality in later life. <em>Genome biology</em>, 16(1):1–12, 2015. URL: <a class="reference external" href="https://doi.org/10.1186/s13059-015-0584-6">https://doi.org/10.1186/s13059-015-0584-6</a>.</p>
</dd>
<dt class="label" id="id59"><span class="brackets"><a class="fn-backref" href="#id20">21</a></span></dt>
<dd><p>Riccardo E Marioni, Sonia Shah, Allan F McRae, Stuart J Ritchie, Graciela Muniz-Terrera, Sarah E Harris, Jude Gibson, Paul Redmond, Simon R Cox, Alison Pattie, and others. The epigenetic clock is correlated with physical and cognitive fitness in the lothian birth cohort 1936. <em>International journal of epidemiology</em>, 44(4):1388–1396, 2015. URL: <a class="reference external" href="https://doi.org/10.1093/ije/dyu277">https://doi.org/10.1093/ije/dyu277</a>.</p>
</dd>
<dt class="label" id="id51"><span class="brackets"><a class="fn-backref" href="#id16">22</a></span></dt>
<dd><p>David H Meyer and Björn Schumacher. Bit age: a transcriptome-based aging clock near the theoretical limit of accuracy. <em>Aging cell</em>, 20(3):e13320, 2021. URL: <a class="reference external" href="https://doi.org/10.1111/acel.13320">https://doi.org/10.1111/acel.13320</a>.</p>
</dd>
<dt class="label" id="id43"><span class="brackets"><a class="fn-backref" href="#id4">23</a></span></dt>
<dd><p>Alexey Moskalev. <em>Biomarkers of human aging</em>. Volume 10. Springer, 2019. URL: <a class="reference external" href="https://link.springer.com/book/10.1007/978-3-030-24970-0">https://link.springer.com/book/10.1007/978-3-030-24970-0</a>.</p>
</dd>
<dt class="label" id="id69"><span class="brackets"><a class="fn-backref" href="#id30">24</a></span></dt>
<dd><p>Paul G Nelson, Daniel EL Promislow, and Joanna Masel. Biomarkers for aging identified in cross-sectional studies tend to be non-causative. <em>The Journals of Gerontology: Series A</em>, 75(3):466–472, 2020. URL: <a class="reference external" href="https://doi.org/10.1093/gerona/glz174">https://doi.org/10.1093/gerona/glz174</a>.</p>
</dd>
<dt class="label" id="id70"><span class="brackets"><a class="fn-backref" href="#id6">25</a></span></dt>
<dd><p>Domenico Palumbo, Ornella Affinito, Antonella Monticelli, and Sergio Cocozza. Dna methylation variability among individuals is related to cpgs cluster density and evolutionary signatures. <em>BMC genomics</em>, 19(1):1–9, 2018. URL: <a class="reference external" href="https://doi.org/10.1186/s12864-018-4618-9">https://doi.org/10.1186/s12864-018-4618-9</a>.</p>
</dd>
<dt class="label" id="id55"><span class="brackets"><a class="fn-backref" href="#id17">26</a></span></dt>
<dd><p>Evgeny Putin, Polina Mamoshina, Alexander Aliper, Mikhail Korzinkin, Alexey Moskalev, Alexey Kolosov, Alexander Ostrovskiy, Charles Cantor, Jan Vijg, and Alex Zhavoronkov. Deep biomarkers of human aging: application of deep neural networks to biomarker development. <em>Aging (Albany NY)</em>, 8(5):1021, 2016. URL: <a class="reference external" href="https://doi.org/10.18632/aging.100968">https://doi.org/10.18632/aging.100968</a>.</p>
</dd>
<dt class="label" id="id66"><span class="brackets"><a class="fn-backref" href="#id27">27</a></span></dt>
<dd><p>Nicholas J Schork, Brett Beaulieu-Jones, Winnie Liang, Susan Smalley, and Laura H Goetz. Does modulation of an epigenetic clock define a geroprotector? <em>Advances in geriatric medicine and research</em>, 2022. URL: <a class="reference external" href="https://doi.org/10.20900%2Fagmr20220002">https://doi.org/10.20900%2Fagmr20220002</a>.</p>
</dd>
<dt class="label" id="id100"><span class="brackets"><a class="fn-backref" href="#id16">28</a></span></dt>
<dd><p>Andrei E Tarkhov, Ramani Alla, Srinivas Ayyadevara, Mikhail Pyatnitskiy, Leonid I Menshikov, Robert J Shmookler Reis, and Peter O Fedichev. A universal transcriptomic signature of age reveals the temporal scaling of caenorhabditis elegans aging trajectories. <em>Scientific reports</em>, 9(1):1–18, 2019. URL: <a class="reference external" href="https://doi.org/10.1038/s41598-019-43075-z">https://doi.org/10.1038/s41598-019-43075-z</a>.</p>
</dd>
<dt class="label" id="id99"><span class="brackets"><a class="fn-backref" href="#id5">29</a></span></dt>
<dd><p>Alexandre Trapp, Csaba Kerepesi, and Vadim N Gladyshev. Profiling epigenetic age in single cells. <em>Nature Aging</em>, 1(12):1189–1201, 2021. URL: <a class="reference external" href="https://doi.org/10.1038/s43587-021-00134-3">https://doi.org/10.1038/s43587-021-00134-3</a>.</p>
</dd>
<dt class="label" id="id57"><span class="brackets"><a class="fn-backref" href="#id14">30</a></span></dt>
<dd><p>Xiaonan Yang, Ling Kui, Min Tang, Dawei Li, Kunhua Wei, Wei Chen, Jianhua Miao, and Yang Dong. High-throughput transcriptome profiling in drug and biomarker discovery. <em>Frontiers in genetics</em>, 11:19, 2020. URL: <a class="reference external" href="https://doi.org/10.3389/fgene.2020.00019">https://doi.org/10.3389/fgene.2020.00019</a>.</p>
</dd>
<dt class="label" id="id49"><span class="brackets"><a class="fn-backref" href="#id13">31</a></span></dt>
<dd><p>Qian Zhang, Costanza L Vallerga, Rosie M Walker, Tian Lin, Anjali K Henders, Grant W Montgomery, Ji He, Dongsheng Fan, Javed Fowdar, Martin Kennedy, and others. Improved precision of epigenetic clock estimates across tissues and its implication for biological ageing. <em>Genome medicine</em>, 11(1):1–11, 2019. URL: <a class="reference external" href="https://doi.org/10.1186/s13073-019-0667-1">https://doi.org/10.1186/s13073-019-0667-1</a>.</p>
</dd>
</dl>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./ml"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="../stat/hazard_models.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">3. </span>Hazard models</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="../dyn/complex_systems.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">5. </span>Complex systems approach</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Kriukov Dmitrii<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>