
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Senescent cell turnover slowdown with age providing an explanation for the Gompertz law &#8212; Computational aging book</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css" />
    <link rel="stylesheet" href="../../../_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../../../" id="documentation_options" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../../../_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="shortcut icon" href="../../../_static/symbol.png"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="Modeling Rate Of Senescence" href="../levine2013/report.html" />
    <link rel="prev" title="Transcriptomics-based longevity drug discovery" href="../drugdiscovery/report.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../../../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../../../_static/symbol.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Computational aging book</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../../intro.html">
                    Welcome to the Computational Biology of Aging coursebook
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Biology of Aging
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../bio/intro_aging_biology.html">
   1. Introduction to Aging Biology
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Statistical analysis
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../stat/differential_analysis_practice.html">
   2. Differential expression analysis
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../../meth/meth.html">
   3. Omics Data Analysis in Aging: Methylomics
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://colab.research.google.com/drive/1mCak-0KTZkISR7ptyBvIOBj1MpewOwDx?usp=sharing">
   Practice - WGCNA for Methylomics
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Biomarkers of Aging
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../stat/survival_analysis.html">
   4. Survival analysis
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://colab.research.google.com/drive/1Cfm2kqMho_AW9voehcP0F2RkvDOXTGFC?usp=sharing">
   Practice - Survival analysis with ML
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../../stat/survival_analysis_part2.html">
   5. Advanced survival analysis
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://colab.research.google.com/drive/13QFIdNcTIYlApM1FdX2uPru34oZ17JLR?usp=sharing">
   Practice - Deep survival analysis
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../../ml/aging_clocks.html">
   6. Aging Clocks
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference external" href="https://colab.research.google.com/drive/1h1D35mKJ1-7YlBRW7tLAIO2mviHIAMuj?usp=sharing">
   Practice - DIY Aging Clocks
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Models of Aging
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../dyn/complex_systems.html">
   7. Complex systems approach
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../../dyn/system_resilience.html">
   8. System resilience
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Student projects 2023
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../../2023_1.html">
   Projects 2023 (1-st wave)
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../dosi/report.html">
     Dynamic Organism State Indicator
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../meta/report.html">
     Meta-analysis of aging transcriptomes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../parkinson/report.html">
     Computational Model of Parkinson’s Disease
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../unsupervized/report.html">
     Unsupervised aging
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="../../2023_2.html">
   Projects 2023 (2-nd wave)
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="../drugdiscovery/report.html">
     Transcriptomics-based longevity drug discovery
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     Senescent cell turnover slowdown with age providing an explanation for the Gompertz law
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../levine2013/report.html">
     Modeling Rate Of Senescence
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../autogpt/report.html">
     AutoInsight: AI-Powered Article Analysis Project
    </a>
   </li>
  </ul>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/ComputationalAgingLab/computational_aging_course"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/ComputationalAgingLab/computational_aging_course/issues/new?title=Issue%20on%20page%20%2Fprojects/2023/gompertz/report.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../../../_sources/projects/2023/gompertz/report.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.md</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#introduction">
   Introduction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#results">
   Results
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#research-workflow-evaluation">
     Research workflow evaluation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#modeling-and-parameter-optimisation">
     Modeling and parameter optimisation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#survival-curves-for-mice">
     Survival curves for mice
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#simulation-of-drosophila-and-c-elegans-survival-hazard-curves">
     Simulation of Drosophila and C. elegans survival/hazard curves.
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#discussion">
   Discussion
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#credits-and-contributions">
   Credits and contributions
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Senescent cell turnover slowdown with age providing an explanation for the Gompertz law</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#introduction">
   Introduction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#results">
   Results
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#research-workflow-evaluation">
     Research workflow evaluation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#modeling-and-parameter-optimisation">
     Modeling and parameter optimisation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#survival-curves-for-mice">
     Survival curves for mice
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#simulation-of-drosophila-and-c-elegans-survival-hazard-curves">
     Simulation of Drosophila and C. elegans survival/hazard curves.
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#discussion">
   Discussion
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#credits-and-contributions">
   Credits and contributions
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="senescent-cell-turnover-slowdown-with-age-providing-an-explanation-for-the-gompertz-law">
<h1>Senescent cell turnover slowdown with age providing an explanation for the Gompertz law<a class="headerlink" href="#senescent-cell-turnover-slowdown-with-age-providing-an-explanation-for-the-gompertz-law" title="Permalink to this headline">#</a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">#</a></h2>
<p>The aging process is a complex phenomenon that involves a multitude of biological changes at the cellular and molecular levels. One of the hallmarks of aging is the accumulation of senescent cells, which are cells that have undergone cell cycle arrest and changes that contribute to tissue dysfunction and degeneration, such as senescence-associated secretory phenotype. The Gompertz law describes the exponential increase in mortality rates with age, but the underlying mechanisms behind this phenomenon are not well understood.</p>
<p>We were tasked to examine this paper by Karin et al. [1], where the authors endeavoured to tie together aging and the accumulation of senescent cells in mice based on a longitudinal dataset. They used mathematical modeling to ascertain the most likely relationship between senescent cells accumulation and age, and verified these inferences by both experiments and simulation.</p>
<p>The team succeded in reproducing the computational parts of the the research in question, including the dynamic model. From that, the dynamics of first passage time in relation to Gompertz law have been outlined.</p>
</section>
<section id="results">
<h2>Results<a class="headerlink" href="#results" title="Permalink to this headline">#</a></h2>
<section id="research-workflow-evaluation">
<h3>Research workflow evaluation<a class="headerlink" href="#research-workflow-evaluation" title="Permalink to this headline">#</a></h3>
<p>The research has been based on the premise that cells transition into senescence from randomly accumualted damage to DNA and telomeres. it is, therefore, carried in the paradygm of the stochastic theory of aging [2].</p>
<p>The concentrations of senescent cells were measured in vivo in mice, who have been genetically modified with a luminescent reporter for the expression of P16 protein, which is one of the biomarkers of senescent cells. P16 protein is the keystone protein in the stalling of cell cycle at the G1/S junction that integrates with Rb and CDK4/6. It has been shown to induce senescence, especially after exposure to ionizing radiation or other DNA damaging agents [3].</p>
<p><img alt="p16" src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3288293/bin/nihms-336168-f0001.jpg" /></p>
<!-- ```{figure} imgs/analysis_schema.png
---
width: 90%
name: analysis_schema
---
The overall schema of transcriptome-based treatments selections
``` -->
<p><strong>P16 protein interactions, reproduced from [3]</strong></p>
<p>The main issue with experimental design is the choice of the experiment the authors used to verify their model - induce senescent cells in the organism and then measure its removal. The approach selected was poorly justified: they employed inflammation induced senescence in lungs by DNA-damagine compound bleomycine. No argumentation was provided for this approach in comparison to other organs or tissues, total body induced senescence or injection of senescent cells, which are other well-estblished approaches [4]. Moreover, the protocol they employed was a source of imprecision, since single dose induction of lung damage by bleomycin has been observed to be revercible after 28 days, and one of the timepoints was 35 days after induction. To combat this, a multiple dosage protocol should be followed [5]. The issue at hand has, unfortunately, not been picked up by the reviewers.</p>
</section>
<section id="modeling-and-parameter-optimisation">
<h3>Modeling and parameter optimisation<a class="headerlink" href="#modeling-and-parameter-optimisation" title="Permalink to this headline">#</a></h3>
<p>The provided data of SnC concentrations through time has been converted to a set of trajectories.</p>
<p>Authors of the article used specialed library in Mathematica to simulate Ito process. In contrast, we implemented Ito process (with reflections on the x-axis to adhere to the biological reality that the fraction of a cell type never drops below zero) completely in Python using Cython library to speed up calculations up to 50 times.</p>
<p>For each subtrajectory the probability of reaching the value on the right side of the subtrajectory was obtained by simulation of Ito process from the left side of each subtajectory. Then logged probability was summed up by all subtrajectories and calculated over 20 thousand points grid of possible parameters each ranging from <span class="math notranslate nohighlight">\(e^{-10}\)</span> to <span class="math notranslate nohighlight">\(e^1\)</span>. Suboprimal parameters were found pretty close to ones obtained by the authors</p>
</section>
<section id="survival-curves-for-mice">
<h3>Survival curves for mice<a class="headerlink" href="#survival-curves-for-mice" title="Permalink to this headline">#</a></h3>
<p>With the previously obtained parameters survival curves were simulated by considering first-time-passage of the threshold of the SnCs for the SnC curves simulated with Ito process (starting at 1 SnC for all mice). The usage of first-time passage model as the statistical framework for the simulation stems from the authors’ adherence to stochastic theory of aging. First-passage time statistic is mathematical approach to make conclusions about stochastic processes and measure mean time to a threshold. The biological sense they entered into this model was that the organism dies after its senescent cells outnumber the immune ones and can not be removed.</p>
</section>
<section id="simulation-of-drosophila-and-c-elegans-survival-hazard-curves">
<h3>Simulation of Drosophila and C. elegans survival/hazard curves.<a class="headerlink" href="#simulation-of-drosophila-and-c-elegans-survival-hazard-curves" title="Permalink to this headline">#</a></h3>
<p>For this paper, the author used data from Drosophila [6] and C.elegans [7] to show SR model can go beyond SnCs to explain the effects of lifespan-modulating interventions.</p>
<p><strong>Step 1</strong>
We obtained the optimal parameters of SR model in Drosophila and C. elegans from this paper. We then entered these values into the SR model to simulate the data of 4000 individuals and determine the life status through the threshold Xc.</p>
<p><strong>Step 2</strong>
We used the Kaplan-Meier estimator to fit the data generated in step 1 and plot survival curves. Additionally, Weibull model to fit the same data and plot hazard curves was employed.</p>
<p><img alt="flies hazard curve" src="https://drive.google.com/uc?id=1Zuwynbk50gahoFOp3gfSX-Aq8cLXO9cH" />
<img alt="flies survival curve" src="https://drive.google.com/uc?id=1NiRIYFyh0L7MHk62GHMtjkvgCUq84dyX" /></p>
<p><strong>Drosophila</strong></p>
<p><img alt="C.elegans hazard curve" src="https://drive.google.com/uc?id=1FlD0Cd1siwy2hY3yr3-OOkiAt3-nin7v" />
<img alt="C.elegans survival curve" src="https://drive.google.com/uc?id=1POK2lAbpIMY6PQ1X4PcyZYxKS0Q0CXH4" /></p>
<p><strong>C.elegans</strong></p>
</section>
</section>
<section id="discussion">
<h2>Discussion<a class="headerlink" href="#discussion" title="Permalink to this headline">#</a></h2>
<p>The authors of the paper have succumbed to an easily avoidable failing in experimental design, which however is minor and is incapable of seriously altering the conclusions of the research.</p>
<p>One important question in the paper is how they derived the Gompertz law from SR model dynamics? To answer this question, we looked at the description of the Gompertz mortality derivation in the SR model.</p>
<p><img alt="Derivation of Gompertz mortality in the SR model" src="https://drive.google.com/uc?id=1EVRD9D9Fj1baaxqDKSbMfhqAHeNlUTYf" /></p>
<p>From the Supplementary Equation 9, we can see the hazard rises exponentially with time. But when ηt &gt; β, this approximation begins to be inaccurate and simulations of the full SR model are needed to compute the hazard curve at old ages. We can derive Equation 9, which we think may have a small error(no +1) for the empirical hazard.<img alt="empirical hazard" src="https://drive.google.com/uc?id=1fcNWNttk58YTFR102PoAvuxuqdo1xBnS" /></p>
<p>In the simulated survival curve analysis of Drosophila and C. elegans, β, κ and ε were initially set, and there may be certain problems in looking only for eta. So a better approach is to perform analysis with the methodology used for mice. Experimental data on Drosophila cannot be obtained with the paper and had to be obtained by corresponding with the authors. Still, the process leading to the smooth curves of the two FF to LE transitions is unclear from the paper.</p>
<p><img alt="flies experiment data hazard curve" src="https://drive.google.com/uc?id=1B5_aCmDAFYsMbGnWUikRMhikLbQxplpG" />
<img alt="flies experiment data survival curve" src="https://drive.google.com/uc?id=1n0g47-78G41dh3Hb95aQTiq8n6_sevi6" /></p>
<p><strong>Drosophila</strong></p>
</section>
<section id="credits-and-contributions">
<h2>Credits and contributions<a class="headerlink" href="#credits-and-contributions" title="Permalink to this headline">#</a></h2>
<p>Voropaev Ivan carried out parameter search and explored the model, Busygin Sergei examined experimental setup and molecular biological background of the study, Song Zhenzhen performed simulations on worm and fly survival curves. All authors equally contributed to writing of this manuscript.</p>
</section>
<section id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">#</a></h2>
<p>[1]	O. Karin, A. Agrawal, Z. Porat, V. Krizhanovsky, and U. Alon, “Senescent cell turnover slows with age providing an explanation for the Gompertz law,” Nat Commun, vol. 10, no. 1, p. 5495, Dec 2 2019, doi: 10.1038/s41467-019-13192-4.</p>
<p>[2]	D. Munoz-Espin and M. Serrano, “Cellular senescence: from physiology to pathology,” Nat Rev Mol Cell Biol, vol. 15, no. 7, pp. 482-96, Jul 2014, doi: 10.1038/nrm3823.</p>
<p>[3]	H. Rayess, M. B. Wang, and E. S. Srivatsan, “Cellular senescence and tumor suppressor gene p16,” Int J Cancer, vol. 130, no. 8, pp. 1715-25, Apr 15 2012, doi: 10.1002/ijc.27316.</p>
<p>[4]	N. Cai, Y. Wu, and Y. Huang, “Induction of Accelerated Aging in a Mouse Model,” Cells, vol. 11, no. 9, Apr 22 2022, doi: 10.3390/cells11091418.</p>
<p>[5]	A. L. Degryse et al., “Telomerase deficiency does not alter bleomycin-induced fibrosis in mice,” Exp Lung Res, vol. 38, no. 3, pp. 124-34, Apr 2012, doi: 10.3109/01902148.2012.658148.</p>
<p>[6]	W. Mair, P. Goymer, S. D. Pletcher, and L. Partridge, “Demography of dietary restriction and death in Drosophila,” Science, vol. 301, no. 5640, pp. 1731-3, Sep 19 2003, doi: 10.1126/science.1086016.</p>
<p>[7]	N. Stroustrup et al., “The temporal scaling of Caenorhabditis elegans ageing,” Nature, vol. 530, no. 7588, pp. 103-7, Feb 4 2016, doi: 10.1038/nature16550.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./projects/2023/gompertz"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="../drugdiscovery/report.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Transcriptomics-based longevity drug discovery</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="../levine2013/report.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Modeling Rate Of Senescence</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Kriukov Dmitrii<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>